src/components/toast/toast.test.tsx

```diff
@@ -0,0 +1,643 @@
+        Child
+      </Toast>
+    );
+
+    expect(screen.getByTestId("toast")).toHaveStyle({ maxWidth: "200px" });
+  });
+
+  it("should render with correct styling when `variant` prop is set to 'notice' and no `onDismiss` prop set", () => {
+    render(
+      <Toast open variant="notice" data-role="toast">
+        foobar
+      </Toast>
+    );
+
+    const toast = screen.getByTestId("toast");
+
+    expect(screen.getByTestId("toast-wrapper")).toHaveStyle({
+      display: "block",
+    });
+    expect(toast).toHaveStyle({
+      "background-color": "var(--colorsUtilityMajor400)",
+      border: "none",
+      color: "var(--colorsSemanticNeutralYang100)",
+      "margin-right": "0",
+      "max-width": "100%",
+    });
+    expect(toast).toHaveStyleRule("right", "55px", {
+      modifier: `${StyledIconButton}`,
+    });
+    expect(screen.getByTestId("toast-content")).toHaveStyle({
+      display: "flex",
+      "align-items": "center",
+      padding: "11px 40px",
+    });
+    expect(screen.queryByTestId("toast-type-icon")).not.toBeInTheDocument();
+  });
+
+  it("should render with correct styling when `variant` prop is set to 'notice' and `onDismiss` prop set", () => {
+    render(
+      <Toast open variant="notice" data-role="toast" onDismiss={() => {}}>
+        foobar
+      </Toast>
+    );
+
+    expect(screen.getByTestId("toast-content")).toHaveStyle({
+      display: "flex",
+      "align-items": "center",
+      padding: "11px 88px 11px 40px",
+    });
+  });
+
+  it("should render with correct styling when `variant` prop is set to 'notice' and `alignY` prop is set to 'top'", () => {
+    render(
+      <Toast
+        open
+        variant="notice"
+        alignY="top"
+        data-role="toast"
+        onDismiss={() => {}}
+      >
+        foobar
+      </Toast>
+    );
+
+    const toast = screen.getByTestId("toast");
+
+    expect(toast).toHaveStyle({
+      "margin-top": "0",
+    });
+  });
+
+  it("should render with correct styling when `variant` prop is set to 'notice' and `alignY` prop is set to 'bottom'", () => {
+    render(
+      <Toast
+        open
+        variant="notice"
+        alignY="bottom"
+        data-role="toast"
+        onDismiss={() => {}}
+      >
+        foobar
+      </Toast>
+    );
+
+    const toast = screen.getByTestId("toast");
+
+    expect(toast).toHaveStyle({
+      "margin-bottom": "0",
+    });
+  });
+
+  it("should render with correct styling when `variant` prop is not set to 'notice' and `alignY` prop is set to 'bottom'", () => {
+    render(
+      <Toast
+        open
+        variant="neutral"
+        alignY="bottom"
+        data-role="toast"
+        onDismiss={() => {}}
+      >
+        foobar
+      </Toast>
+    );
+
+    const toast = screen.getByTestId("toast");
+
+    expect(toast).toHaveStyle({
+      "margin-bottom": "30px",
+    });
+  });
+
+  it("should render the expected icon when `variant` prop is set to 'neutral'", () => {
```

Comment:
The intention was to move away from excessively nesting `describe` and `it` blocks:

```tsx
describe("block 1", () => {
   describe("block 2", () => {
      it("test", () => {
      
      });
   })
});

as it increases complexity and makes the test file harder to read. Also since we were declaring variables and rendering content in outer scopes, this increased the risk of tests not being properly isolated.

Using parameterised tests (or some describe blocks) is ok if it helps improve readability :)
```

Reply:
I thought we wanted to avoid iteration in favour of explicitly defining a test case

Reply:
Good point about having too many parameters 👍🏼 In that case, I'm onboard with sticking to the current approach :)